tests/ui/impl-trait/failed-to-resolve-instance-ice-123145.rs

```diff
@@ -1,7 +1,5 @@
 // ICE failed to resolve instance for ...
 // issue: rust-lang/rust#123145
-//@ build-fail
```

Comment:
This now cycles in `check_opaque_well_formed`, since now we actually reveal `impl Handler` when proving that it satisfies its own item bounds. This seems desirable to me.